pydust/src/types/obj.zig

```diff
@@ -33,45 +33,20 @@ pub const PyObject = extern struct {
         return name.asSlice();
     }
 
-    /// Call this object without any arguments.
-    pub fn call0(self: PyObject) !PyObject {
-        return .{ .py = ffi.PyObject_CallNoArgs(self.py) orelse return PyError.PyRaised };
-    }
-
     /// Call this object with the given args and kwargs.
-    pub fn call(self: PyObject, args: anytype, kwargs: anytype) !PyObject {
-        var argsPy: py.PyTuple = undefined;
-        if (@typeInfo(@TypeOf(args)) == .Optional and args == null) {
-            argsPy = try py.PyTuple.new(0);
-        } else {
-            argsPy = try py.PyTuple.checked(try py.create(args));
-        }
-        defer argsPy.decref();
-
-        // FIXME(ngates): avoid creating empty dict for kwargs
-        var kwargsPy: py.PyDict = undefined;
-        if (@typeInfo(@TypeOf(kwargs)) == .Optional and kwargs == null) {
-            kwargsPy = try py.PyDict.new();
-        } else {
-            const kwobj = try py.create(kwargs);
-            if (try py.len(kwobj) == 0) {
-                kwobj.decref();
-                kwargsPy = try py.PyDict.new();
-            } else {
-                kwargsPy = try py.PyDict.checked(kwobj);
-            }
-        }
-        defer kwargsPy.decref();
-
-        // We _must_ return a PyObject to the user to let them handle the lifetime of the object.
-        const result = ffi.PyObject_Call(self.py, argsPy.obj.py, kwargsPy.obj.py) orelse return PyError.PyRaised;
-        return PyObject{ .py = result };
+    pub fn call(self: PyObject, comptime T: type, method: [:0]const u8, args: anytype, kwargs: anytype) !T {
+        const methodObj = try self.get(method);
+        return py.call(T, methodObj, args, kwargs);
```

Comment:
there's C API for method call fwiw

Reply:
[PyObject_CallMethodObjArgs](https://docs.python.org/3/c-api/call.html#c.PyObject_CallMethodObjArgs) but I would check if it's actually different/better

Reply:
eh, nope. It doesn't handle kwargs.